### PR TITLE
Migrate OAuth login to new package

### PR DESCRIPTION
**Pull request by e-wai**
Screen Shot 2022-12-27 at 1 53 29 AM

## Implementation description
* Using this new package: https://www.npmjs.com/package/@react-oauth/google
  * Sample - https://github.com/MomenSherif/react-oauth/blob/47ec50f7f6b32d96f20cac63b21533ae74b49e3b/apps/playground/src/components/SignInFlows.tsx#L40-L47
  * Styles - https://github.com/MomenSherif/react-oauth/blob/8ed49b8ef6cd6f59fd11c233fe8157a381413076/packages/%40react-oauth/google/src/types/index.ts#L70 
* https://developers.google.com/identity/gsi/web/guides/migration

## Steps to test
1. Log in with bp email
2. See error state from logging in with non-bp email

## What should reviewers focus on?
* It's not the right colour, but the styles didn't have it as an option. Didn't think it was worth building completely from scratch

Screen Shot 2022-12-27 at 1 57 11 AM

Screen Shot 2022-12-27 at 1 57 50 AM

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: .github/actions/heroku-deploy/action.yml [hunk 1]
```diff
@@ -91,7 +93,8 @@ runs:
         heroku config:set MAILER_REFRESH_TOKEN="${{ inputs.mailer-refresh-token }}" -a $HEROKU_APP_NAME && \
         heroku config:set STRIPE_PUBLISHABLE_TEST_KEY="${{ inputs.stripe-publishable-key }}" -a $HEROKU_APP_NAME && \
         heroku config:set STRIPE_SECRET_TEST_KEY="${{ inputs.stripe-secret-key }}" -a $HEROKU_APP_NAME && \
-        heroku config:set PREVIEW_DEPLOY=true -a $HEROKU_APP_NAME
+        heroku config:set PREVIEW_DEPLOY=$([ "${{inputs.deploy-branch}}" = "main" ] && echo "false" || echo "true") -a $HEROKU_APP_NAME
```
Comment (e-wai) [inline .github/actions/heroku-deploy/action.yml:96]: https://github.com/uwblueprint/focus-on-nature/blob/2badea11410114fd5c3838b6272d02e23dffe0b1/backend/typescript/rest/authRoutes.ts#L22-L26

**Comment by e-wai** [comment]
(Preview link looks right on login page, but request fails because the origin/domain isn't added to allowlist)

File: frontend/yarn.lock [hunk 2]
```diff
@@ -2413,6 +2413,11 @@
   resolved "https://registry.yarnpkg.com/@popperjs/core/-/core-2.11.5.tgz#db5a11bf66bdab39569719555b0f76e138d7bd64"
   integrity sha512-9X2obfABZuDVLCgPK9aX0a/x4jaOEweTTWE2+9sr0Qqqevj2Uv5XorvusThmc9XGYpS9yI+fhh8RTafBtGposw==
 
+"@react-oauth/google@^0.5.1":
```
Comment (dhmi19) [inline frontend/yarn.lock:2416]: Are we pushing yarn.lock files to github? I think bowen had some dependancy changes but didn't push the yarn.lock file. If we're putting it on the repo, then I think we should add his yarn.lock file before merging his pr too

Comment (e-wai) [reply frontend/yarn.lock:2416]: Yeah don't see any reason not to

**Review by dhmi19 — APPROVED** [review]
thankssssssss

File: backend/typescript/rest/authRoutes.ts [hunk 3]
```diff
@@ -28,10 +28,10 @@ const cookieOptions: CookieOptions = {
 /* Returns access token and user info in response body and sets refreshToken as an httpOnly cookie */
 authRouter.post("/login", loginRequestValidator, async (req, res) => {
   try {
-    const authDTO = req.body.idToken
-      ? // OAuth
-        await authService.generateTokenOAuth(req.body.idToken)
-      : await authService.generateToken(req.body.email, req.body.password);
```
[the remark this answers is not in the document]
Comment (e-wai) [reply backend/typescript/rest/authRoutes.ts:34]: No, we have no login flow with email / password

**Pull request merged**